### PR TITLE
Avoid kubel from always moving the point to point-min when refreshing

### PR DESCRIPTION
**Pull request by Konubinix**
tabulated-list already provides a way to remember the position between refreshes. It only need to be used.

### CODE REVIEW

**Comment by abrochard** [comment]
Thank you for submitting! I didn't know about this feature.
Does this mean that [`kubel--save-line`](https://github.com/abrochard/kubel/blob/master/kubel.el#L592) and [`kubel--jump-back-to-line`](https://github.com/abrochard/kubel/blob/master/kubel.el#L599) and stuff can be deleted?

**Comment by Konubinix** [comment]
Adrien Brochard ***@***.***> writes:

> Thank you for submitting! I didn't know about this feature.
> Does this mean that [`kubel--save-line`](https://github.com/abrochard/kubel/blob/master/kubel.el#L592) and [`kubel--jump-back-to-line`](https://github.com/abrochard/kubel/blob/master/kubel.el#L599) and stuff can be deleted?

Interesting, I did not know about those. Somehow my use of kubel
bypassed it.

I suppose that depends on your preference. Both behavior make sense to
me. kubel--save-line saves make the cursor look static, while the
content might change. (tabulated-list-print t) makes the cursor follow
the content, even if it is moved.

Supporting both could be a good thing, but add maintenance cost.

What do you think?

-- 
Konubinix
GPG Key    : 7439106A
Fingerprint: 5993 BE7A DA65 E2D9 06CE  5C36 75D2 3CED 7439 106A

**Comment by abrochard** [comment]
I like your version better. If my cursor is on a pod and refresh, I would like my cursor still on that same pod.
What would the behavior be if the item disappears? Like I have my cursor on a terminating pod and it is gone after a refresh? Would the cursor stay at the same line or go back to the top?

**Comment by Konubinix** [comment]
Adrien Brochard ***@***.***> writes:

> I like your version better. If my cursor is on a pod and refresh, I would like my cursor still on that same pod.
> What would the behavior be if the item disappears? Like I have my cursor on a terminating pod and it is gone after a refresh? Would the cursor stay at the same line or go back to the top?

If I remember correctly, it goes back to the top.

-- 
Konubinix
GPG Key    : 7439106A
Fingerprint: 5993 BE7A DA65 E2D9 06CE  5C36 75D2 3CED 7439 106A

**Comment by Konubinix** [comment]
Actually, I found out that my issue with the line moving back to the top was due to some conflict between window-point and point.

Now, I may change this PR to use the default behavior of tabulate-print while falling back to recording the line.

In the mean time, I suggested another PR to make the code that records the line number shorter and easier to read (IMHO).

**Comment by abrochard** [comment]
Hmm. I think I'd need to test it out. 
I really like that tabulated list offers the functionality and it makes sense to track the "value" and not the line. But if I'm observing a terminating pod and refresh and my cursor goes back to the first line, that's not a great experience.

**Comment by Konubinix** [comment]
Actually, it was easier to do than I expected, so I integrated the changes of #102 into this PR and closed it.

Now, this PR has the behavior that I would expect from kubel: it keeps track of the current item and falls back on keeping the same line if the current item vanishes.

It also replaces the kubel--save-line and kubel--jump-back-to-line logic by something that is shorter and (IMHO) is more readable.

**Comment by abrochard** [comment]
Wondeful! Thank you.
I'll review and test.

File: kubel.el [hunk 1]
```diff
@@ -1327,11 +1313,15 @@ DIRECTORY is optional for TRAMP support."
...
   (setq tabulated-list-sort-key nil)
   (tabulated-list-init-header)
-  (tabulated-list-print)
+  (let ((line-num (line-number-at-pos (point)))
+        (current-id (tabulated-list-get-id)))
+    (tabulated-list-print t)
+    (unless (string-equal current-id (tabulated-list-get-id))
+      ;; tabulated-list could not follow the current entry, then fallback on
+      ;; keeping the same line.
+      (goto-line line-num)))
```
Comment (abrochard) [inline kubel.el:1322]: I thought that linter complained about `goto-line` usage in non-interactive? See #101

File: kubel.el [hunk 2]
```diff
@@ -664,19 +664,6 @@ TYPENAME is the resource type/name."
...
-(defun kubel--save-line ()
-  "Save the current line number if the view is unchanged."
-  (if (equal (buffer-name (current-buffer))
-             (kubel--buffer-name))
-      (setq kubel--line-number (+ 1 (count-lines 1 (point))))
-    (setq kubel--line-number nil)))
-
-(defun kubel--jump-back-to-line ()
-  "Jump back to the last cached line number."
-  (when kubel--line-number
```
Comment (abrochard) [inline kubel.el:676]: can we also delete `kubel--line-number` if it's not used?

**Comment by Konubinix** [comment]
Adrien Brochard ***@***.***> writes:

> @abrochard commented on this pull request.
> can we also delete `kubel--line-number` if it's not used?

Done
    
-- 
Konubinix
GPG Key    : 7439106A
Fingerprint: 5993 BE7A DA65 E2D9 06CE  5C36 75D2 3CED 7439 106A

File: kubel.el (hunk 2 above)
Comment (Konubinix) [reply kubel.el:676]: Done

File: kubel.el (hunk 1 above)
Comment (Konubinix) [reply kubel.el]: Indeed, I fixed that

**Review by abrochard — APPROVED** [review]
Thank you!

**Pull request merged**